### ISSUE TITLE
JDBC client throws exception on unrecognized parameter instead of simply warning and ignoring

### ISSUE THREAD

**Issue by PuszekSE** [trinodb/trino#20301]
I've tried using spark to read data from trino using trino-jdbc client, but the driver fails on 

```
throw new SQLException(format("Unrecognized connection property '%s'", propertyName));
```
Since spark passes sql read as 'dbtable' or 'query' parameter, it's failing

```
.option("dbtable", "(SELECT * FROM <something>) as t")
```

Just so you know, it does work perfectly fine, if you change throwing exceptions to logging warnings and filtering unrecognized properties.

I don't believe that strict fail as exception is viable approach for such trivial thing as unrecognized property.

**Comment by hashhar** [comment]
It's a bugfix in Spark for that already, https://issues.apache.org/jira/browse/SPARK-36163

**Comment by PuszekSE** [comment]
From my perspective, spark was simply an example - I see no value of strict properties' evaluation, but maybe your team deemed it otherwise... Most of the tools I've worked with over the years had warn&ignore-based approach.

Regarding the fix on Spark's side, to be honest, it being fixed in spark 3.3.0+ is an issue by itself, as Spark has a lot of backwards compatibility issues, which makes it obsolete in many cases...
---
Anyways, thanks for responding - for now I'll simply use modified client, as it seems to be valid option from my side.
